server/s3_delete_object_action.cc

```diff
@@ -293,6 +294,39 @@ void S3DeleteObjectAction::delete_metadata_failed() {
   send_response_to_s3_client();
 }
 
+void S3DeleteObjectAction::save_bucket_counters() {
+  s3_log(S3_LOG_INFO, stripped_request_id, "%s Entry\n", __func__);
+  int64_t inc_object_count = 0;
+  int64_t inc_obj_size = 0;
+
+  inc_object_count = -1;
+  inc_obj_size = -(object_metadata->get_content_length());
+
+  S3BucketCapacityCache::update_bucket_capacity(
+      request, bucket_metadata, inc_object_count, inc_obj_size,
+      std::bind(&S3DeleteObjectAction::save_bucket_counters_success, this),
+      std::bind(&S3DeleteObjectAction::save_bucket_counters_failed, this));
```

Comment:
this counter handling and metadata deletion should also be added to rollback sequence -- if some failure we have to put it back into motr.  Or we need to modify BG service to somehow handle this all.

Reply:
To be done in Non-happy path JIRA